plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushHiveMetastoreCacheProcedure.java

```diff
@@ -116,8 +117,13 @@ private void doFlushMetadataCache(Optional<String> schemaName, Optional<String>
         if (schemaName.isEmpty() && tableName.isEmpty() && partitionColumns.isEmpty()) {
             cachingHiveMetastore.flushCache();
         }
-        else if (schemaName.isPresent() && tableName.isPresent() && !partitionColumns.isEmpty()) {
-            cachingHiveMetastore.flushPartitionCache(schemaName.get(), tableName.get(), partitionColumns, partitionValues);
+        else if (schemaName.isPresent() && tableName.isPresent()) {
+            if (!partitionColumns.isEmpty()) {
```

Comment:
It would be nice if we could resolve a call procedure based on the name and arguments. So we could have an individual method for each configuration - instead of if-else.